packages/material-ui/src/utils/useId.js

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+
+/**
+ * Private module reserved for @material-ui/x packages.
```

Comment:
In that case the name should make this obvious e.g. `internal_useId` which we'll import like `import { internal_useId as useId } from '@material-ui/utils'`.

Reply:
Happy to make that change, @oliviertassinari you in agreement also? Or shall we just make this public?

Reply:
No preference, but we should make sure we have a consistent approach with 
https://github.com/mui-org/material-ui/blob/02e47fd2cc12b34b485e2f7b59764d8be30c455c/packages/material-ui/src/utils/createSvgIcon.js#L4-L7

Reply:
Has `createSvgIcon` been released? Because if we change it now, then older version of `lab` could be trying to import a newer version where `createSvgIcon` is renamed to `unstable_createSvgIcon` and would break.

Reply:
I don't know what createSvgIcon has to do with this PR. In my book one correct approach is already better than two incorrect ones. I wouldn't block this one because we need a consistent approach. I  find this debateable for internal code. As long as it works and don't cause churn in the long run I'm fine with it.

Reply:
Okay, I'll leave `createSvgIcon` out of it, I'll keep the comment in `useId` and change it to `unstable_useId`

Reply:
Ok, so I would propose `unstable_useId` and we move forward from this point.

Reply:
Do you want me to also name the module `unstable_useId`, or can we keep the module name as `useId`, but when we export it from `@material-ui/core/utils`, we do `export { default as unstable_useId } from './useId'`. The benefit is that internally in `core`, we don’t need to use the `unstable` prefix

Reply:
I could find this prior-work: #14593. But `unstable_useId` all the way through sounds great.

Reply:
Getting ESLint errors about `unstable_useId`. Would you prefer

```js
// eslint-disable-next-line camelcase
import unstable_useId from '../utils/unstable_useId';
```

Or 

```js
import useId from '../utils/unstable_useId';
```